phd-tests/framework/src/lifecycle.rs

```diff
@@ -0,0 +1,94 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use std::time::Duration;
+
+use tracing::info;
+use uuid::Uuid;
+
+use crate::{Framework, TestVm};
+
+/// The set of actions that can be taken on a VM undergoing lifecycle testing.
+pub enum Action<'a> {
+    /// Reset the VM using the Propolis server reset API. This sort of reboot
+    /// does not involve the guest OS.
+    Reset,
+
+    /// Stop the VM and restart it in a successor Propolis using the same
+    /// environment as its predecessor.
+    StopAndStart,
+
+    /// Migrate the VM to a new Propolis server. The wrapped `&str` names a
+    /// Propolis server artifact to migrate to.
+    MigrateToPropolis(&'a str),
```

Comment:
I won't be the first to cast stones regarding bits of an interface which are sketched out, but not (wholly) yet in use, but it might be worth a little comment to that effect (at least for `Reset` and `MigrateToPropolis`)